site_tools/image_deployer.py

```diff
@@ -30,21 +31,34 @@ def __str__(self):
         return f"Image(name={self.name})"
 
 
-def get_openstack_connection(cloud, cloud_name):
-    return openstack.connect(auth_url=cloud['clouds'][cloud_name]["auth"]["auth_url"],
-                             auth_type="v3applicationcredential",
-                             application_credential_id=cloud['clouds'][cloud_name]["auth"]["application_credential_id"],
-                             application_credential_secret=cloud['clouds'][cloud_name]["auth"]["application_credential_secret"],
-                             region_name=cloud['clouds'][cloud_name]["region_name"])
+def get_openstack_connection(cloud_name):
+    return openstack.connect(cloud=cloud_name)
+
+
+def setup_swift_connection(openstack_conn,
+                           use_object_store_creds=False):
+    auth_token = openstack_conn.auth_token
+    storage_url = openstack_conn.object_store.get_endpoint()
+    swift_conn = SwiftConnection(
+        preauthtoken=auth_token if use_object_store_creds else None,
+        preauthurl=storage_url,
+        auth_version='3'
+    )
```

Comment:
nice! 
I didn't realize the python swift API could be coerced into setting up an auth-less connection, some of the other service libraries are very picky about it :)

Reply:
I'm not entirely sure it is, I'm making a bit of an assumption here and when I remove testing behind the creds of my account we could find out I'm wrong on this. Probably should do that at some point here.

Reply:
if this doesn't work with swift, were you just thinking this would use something like requests to download the images since we don't need auth here?

Reply:
yep, requests or similar for fetching the manifest files, and for now requests for grabbing the image data as well.

[from the other comment, eventually we can skip downloading the image data entirely, and just pass the url to glance-image-import](https://github.com/ChameleonCloud/abracadabra/pull/4#discussion_r1977899027) (once enabled at a given site)

Reply:
Ok, I messed around with swift for a bit and gave up, switched to requests and ripped out swift altogether since it just added clutter